ci/docker/conda-integration.dockerfile

```diff
@@ -25,32 +25,27 @@ ARG node=14
 ARG jdk=8
 ARG go=1.15
 
+# Uninstall unused space-consuming packages
+# (XXX: it would be better not to install them, but they are used by other
+#  builds which are also based on conda-cpp)
+RUN conda uninstall -q clangdev llvmdev valgrind
+
+# Install Archery and integration dependencies
 COPY ci/conda_env_archery.yml /arrow/ci/
 RUN conda install -q \
         --file arrow/ci/conda_env_archery.yml \
         numpy \
         maven=${maven} \
         nodejs=${node} \
         openjdk=${jdk} && \
-    conda clean --all
+    conda clean --all --force-pkgs-dirs
 
-RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+# Install Rust and remove space-consuming docs
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y && \
+    rm -rf `find /root/.rustup/ -name html -type d`
```

Comment:
@alamb Is there a better way to do this?

Reply:
@pitrou  
I wonder if we could try to pass the `--profile=minimal` option to the script (docs at https://rust-lang.github.io/rustup/concepts/profiles.html). 

Perhaps something like 
```
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --profile=minimal -y
```

When I ran that command on a local docker image the rust tool chain takes up around 400MB: 

```
root@46bd64d9d397:/# du -s -h  /root/.rustup/
411M	/root/.rustup/
```

Running 
```
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```

consumes 850MB
```
root@86dbaf9d4c6b:/# du -s -h /root/.rustup 
865M	/root/.rustup
```

Reply:
Oh, cool! I'll try it.

Reply:
It works, thank you!

Reply:
Hmm, actually, it seems `rustfmt` is needed to build Flight support:
https://github.com/pitrou/arrow/runs/2270630220?check_suite_focus=true#step:8:1629

Do you know if that is intended?

Reply:
@pitrou I don't think we use `rustfmt` directly, but rather it is used by `tonic_build` https://docs.rs/tonic-build/0.4.1/tonic_build/ which is used to compile the `protobuf` definitions.